pkg/middleware/scope.go

```diff
@@ -13,7 +13,7 @@ func NewScope(reverseProxy bool) alice.Constructor {
 		return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
 			scope := &middlewareapi.RequestScope{
 				ReverseProxy: reverseProxy,
-				RequestID:    requestutil.GetRequestID(req),
+				RequestID:    requestutil.GenRequestID(req),
```

Comment:
I had intended for the generation logic to be in this package, that way the request ID header that is configured doesn't need to be set anywhere as a package level var. Everything else can rely just on the request scope

Reply:
So I'm torn for 2 reasons:

1.) The actual business logic really matches everything else that is happening in `requestutils`.
2.) If the `requestutils.GetRequestID` we consume throughout the application is truly only `RequestScope` aware, in tests we'd need to setup a full scope setting middleware in all their setup. At the moment it's easy to just splice in the extra header for non-middleware related test suites.

The original intent was to memoize the random UUID case, and I guess things got wonky because we jumped across boundaries to `RequestScope` to perform the semi-memoization (since it was the first caller setting it on itself). In a `RequestScope` free world, I would memoize this in the `GetRequestID` on its own Context key.

Do we want to do that? Or maybe still memoize in the `RequestScope`, but not set it on `RequestScope` initialization, but instead actually memoize (set the stored value on the first real call to it)?

Reply:
Ah - I follow, the concern is mostly around the management of the custom request ID header -- missed that aspect in the previous discussion 🤔 

> This way we can pass the request ID header as a parameter when we construct the middleware scope, will likely be cleaner than the current implementation right?

Reply:
Eh - just realized concern #2 isn't that big of a deal, we have this at our fingertips to setup tests without the full middleware handler: `req = middlewareapi.AddRequestScope(req, scope)`

Reply:
We have created a pattern where we add the request scope in all our tests anyway, so ensuring this always gets created with request ID at the beginning as well shouldn't add too much extra bloat right?

Reply:
Yeah, I think it isn't too bad. I was initially thinking of the handler route where we spun up the whole middleware for all handler related tests. Our non-middleware test suites didn't have that easy injection point. But I can use `req = middlewareapi.AddRequestScope(req, scope)` super easily 👍 

I think my plan of attack is to completely remove request ID logic from `requestutils` and have the logic be owned by `RequestScope`

Reply:
Alrighty - Moved everything over to `RequestScope` management in the last commit and ditched the package variables & `requestutils` management of the ID.